lesson11/index.js

```diff
@@ -0,0 +1,68 @@
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/77w5';
+
+const ul = document.getElementById('js-list');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.style.listStyle = "none";
+    li.id = "js-loading";
+    loadingImg.src = 'loading-circle.gif';
+    ul.appendChild(li).appendChild(loadingImg);
+};
+
+const removeLoading = () => document.getElementById('js-loading').remove();
+
+const addList = (val) => {
+    const fragment = document.createDocumentFragment();
+    for (let i = 0; i < val.length; i++) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+    
+        a.href = `${val[i].a}.html`;
+        a.textContent = `${val[i].text}`;
+        img.src = val[i].img;
+        img.alt = val[i].alt;
+    
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+
+    ul.appendChild(fragment);
+};
+
+/**
+ * Promiseを実行し、resolveの値を返す関数
+ * @return {promise object} 
+ */
+ const fetchData = (json) => {
+    return new Promise( resolve => {
+        setTimeout(() => {
+            resolve(json);
+        }, 3000);
+    });
+};
+
+/**
+ * Async関数
+ * responseで値が返ってきた後、addList関数にその値を渡す
+ */
+const execution = async (url) => {
+    createLoading();
+
+    try {
+        const response = await fetch(url) // 中身はresponseオブジェクトを含むpromise
+        const json = await response.json(); //response オブジェクトから JSON を抽出
+        const data = await fetchData(json.data); //jsonをfetchData関数に渡す
+        addList(data);
+
+    } catch(error) {
+        console.error(error);
+
+    }  finally {
+        removeLoading();
+    }
+};
```

Comment:
After the previous fetchData(), I further executed the try, catch, and finally statements.

```suggestion
    createLoading();
    try {
        const data = await fetchData();
        if (data.length === 0) {
            throw new Error("データが空です");
        }
        return data;
    } catch (e) {
        createErrorMessage(e);
    } finally {
        removeLoading();
    }
```

below is what Monaka-san reviewed for me.Share this.
- If the data is empty, display the error message 'Data is empty'.
- If the fetchData() function catches an error, write the code to display the error in the catch statement.
　It is easy to see if you make it into a function like the following
```
function createErrorMessage(e) {
    const li = document.createElement("li");
    li.textContent = `エラー内容:${e.message}`;
    ul.appendChild(li);
    console.error(e.message);
};
```

Reply:
The addList() function has also been rewritten as async.
Here is a code sandbox where I tried to write the json fetch in the fetchData function.
If you think it's not the same, please feel free to point it out.
[CodeSandbox](https://codesandbox.io/s/lesson11-jsonqu-de-wofetchdataguan-shu-nishu-itemimasita-xbt6s?file=/index.js)

Reply:
Change execution function contents.
[7762888](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/7762888a89a690a395a5aa69642876db68c613d3)

Created createErrorMessage functions.
[989d64f](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/989d64f862f35b0534df2519d38e24b0b5c0f5f3)

Thank you for tell me!💓

Reply:
The AddList function is the only function that adds.
Executing fetchListData within the AddList function is a process that does not fit the function name.
Therefore, we decided to adopt the method of passing arguments to the AddList function.

↓The code that Haru gave me.
```
const addList = async () => {
  const val = await fetchListData();
```

↓The code I came up with
```
const createDataList = async () => {
    createLoading();
    try {
        const data = await fetchData();
        if( data.length === 0 ) {
            throw new Error('データが空です。');
        }
        addList(data);

    } 
```